### PR TITLE
Various refactorings in the rustdoc module.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

**Comment by alexcrichton** [comment]
@bors: r+

Thanks!

File: src/librustdoc/html/format.rs [hunk 1]
```diff
@@ -516,12 +516,12 @@ impl fmt::Display for clean::Type {
                     clean::Vector(ref bt) => { // BorrowedRef{ ... Vector(T) } is &[T]
                         match **bt {
                             clean::Generic(_) =>
-                                primitive_link(f, clean::Slice,
+                                primitive_link(f, PrimitiveType::Slice,
                                     &format!("&amp;{}{}[{}]", lt, m, **bt)),
                             _ => {
-                                primitive_link(f, clean::Slice, &format!("&amp;{}{}[", lt, m))?;
+                                primitive_link(f, PrimitiveType::Slice, &format!("&amp;{}{}[", lt, m))?;
```
Comment (eddyb) [inline src/librustdoc/html/format.rs]: Fails tidy, line too long.

**Comment by frewsxcv** [comment]
`make tidy` fixed in the latest force push.

**Pull request merged**